### ISSUE TITLE
Load error

### ISSUE THREAD

**Issue by Varstahl** [Drahsid/ResizableHUD#1]
I was looking into plugins that could help me properly move and possibly help align/snap hud components around, and stumbled on yours. Problem is, it won't load:

![image](https://user-images.githubusercontent.com/284077/212498497-0ed5fd5e-060d-4c30-ba6d-611100b35c1c.png)

Any suggestion?

**Comment by Nox13last** [comment]
When asking for help with plugins not loading properly, it's generally a good idea to include a **log**. You can get it by typing _/xllog_ in chat, and then pressing the "copy to clipboard" button.
Here's mine.
```
--------------------------------
22:24:05.737 | ERR [LOCALPLUGIN] Error while loading Resizable HUD
System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation.
 ---> System.NullReferenceException: Object reference not set to an instance of an object.
   at ResizableHUD.Plugin..ctor(DalamudPluginInterface pi, CommandManager commands, ChatGui chat, ClientState clientState)
   at System.RuntimeMethodHandle.InvokeMethod(Object target, Void** arguments, Signature sig, Boolean isConstructor)
   at System.Reflection.ConstructorInvoker.Invoke(Object obj, IntPtr* args, BindingFlags invokeAttr)
   --- End of inner exception stack trace ---
   at System.Reflection.ConstructorInvoker.Invoke(Object obj, IntPtr* args, BindingFlags invokeAttr)
   at System.Reflection.RuntimeConstructorInfo.Invoke(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
   at Dalamud.IoC.Internal.ServiceContainer.CreateAsync(Type objectType, Object[] scopedObjects) in C:\goatsoft\companysecrets\dalamud\IoC\Internal\ServiceContainer.cs:line 107
   at Dalamud.Plugin.Internal.Types.LocalPlugin.LoadAsync(PluginLoadReason reason, Boolean reloading) in C:\goatsoft\companysecrets\dalamud\Plugin\Internal\Types\LocalPlugin.cs:line 422
--------------------------------
22:24:05.737 | ERR [PLUGINW] Plugin installer threw an unexpected error
System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation.
 ---> System.NullReferenceException: Object reference not set to an instance of an object.
   at ResizableHUD.Plugin..ctor(DalamudPluginInterface pi, CommandManager commands, ChatGui chat, ClientState clientState)
   at System.RuntimeMethodHandle.InvokeMethod(Object target, Void** arguments, Signature sig, Boolean isConstructor)
   at System.Reflection.ConstructorInvoker.Invoke(Object obj, IntPtr* args, BindingFlags invokeAttr)
   --- End of inner exception stack trace ---
   at System.Reflection.ConstructorInvoker.Invoke(Object obj, IntPtr* args, BindingFlags invokeAttr)
   at System.Reflection.RuntimeConstructorInfo.Invoke(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
   at Dalamud.IoC.Internal.ServiceContainer.CreateAsync(Type objectType, Object[] scopedObjects) in C:\goatsoft\companysecrets\dalamud\IoC\Internal\ServiceContainer.cs:line 107
   at Dalamud.Plugin.Internal.Types.LocalPlugin.LoadAsync(PluginLoadReason reason, Boolean reloading) in C:\goatsoft\companysecrets\dalamud\Plugin\Internal\Types\LocalPlugin.cs:line 422
```

**Comment by Drahsid** [comment]
The plugin works on my end, but given the error my naive theory is that I probably forgot to update the plugin on my repo (though I am sure I did that.) Just in case, I just updated all of the plugins on my repo. Try again and let me know how it goes.

**Comment by Nox13last** [comment]
```
--------------------------------
01:32:02.540 | ERR [LOCALPLUGIN] Error while loading Resizable HUD
System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation.
 ---> System.NullReferenceException: Object reference not set to an instance of an object.
   at ResizableHUD.Plugin..ctor(DalamudPluginInterface pi, CommandManager commands, ChatGui chat, ClientState clientState)
   at System.RuntimeMethodHandle.InvokeMethod(Object target, Void** arguments, Signature sig, Boolean isConstructor)
   at System.Reflection.ConstructorInvoker.Invoke(Object obj, IntPtr* args, BindingFlags invokeAttr)
   --- End of inner exception stack trace ---
   at System.Reflection.ConstructorInvoker.Invoke(Object obj, IntPtr* args, BindingFlags invokeAttr)
   at System.Reflection.RuntimeConstructorInfo.Invoke(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
   at Dalamud.IoC.Internal.ServiceContainer.CreateAsync(Type objectType, Object[] scopedObjects) in C:\goatsoft\companysecrets\dalamud\IoC\Internal\ServiceContainer.cs:line 107
   at Dalamud.Plugin.Internal.Types.LocalPlugin.LoadAsync(PluginLoadReason reason, Boolean reloading) in C:\goatsoft\companysecrets\dalamud\Plugin\Internal\Types\LocalPlugin.cs:line 456
--------------------------------
01:32:02.540 | ERR [PLUGINW] Plugin installer threw an unexpected error
System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation.
 ---> System.NullReferenceException: Object reference not set to an instance of an object.
   at ResizableHUD.Plugin..ctor(DalamudPluginInterface pi, CommandManager commands, ChatGui chat, ClientState clientState)
   at System.RuntimeMethodHandle.InvokeMethod(Object target, Void** arguments, Signature sig, Boolean isConstructor)
   at System.Reflection.ConstructorInvoker.Invoke(Object obj, IntPtr* args, BindingFlags invokeAttr)
   --- End of inner exception stack trace ---
   at System.Reflection.ConstructorInvoker.Invoke(Object obj, IntPtr* args, BindingFlags invokeAttr)
   at System.Reflection.RuntimeConstructorInfo.Invoke(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
   at Dalamud.IoC.Internal.ServiceContainer.CreateAsync(Type objectType, Object[] scopedObjects) in C:\goatsoft\companysecrets\dalamud\IoC\Internal\ServiceContainer.cs:line 107
   at Dalamud.Plugin.Internal.Types.LocalPlugin.LoadAsync(PluginLoadReason reason, Boolean reloading) in C:\goatsoft\companysecrets\dalamud\Plugin\Internal\Types\LocalPlugin.cs:line 456
```

This is on Version 1.0.0.38832

**Comment by Drahsid** [comment]
Huh?? The plugin version in my repo is "1.0.0.33086", this is both in the json and assembly. If you download it from [here](http://repo.modloader64.com/ffxiv/stable/ResizableHUD_Release.zip), can you manually add it?
Edit: One more thing, I may have solved the issue you were getting on our end (possibly just a dns cache issue), see if it is still giving you the incorrect version.

**Comment by Nox13last** [comment]
I can't confirm whether this is the version number I still have, as I deleted it as part of my troubleshooting step. This is what I'm getting upon trying to get it from the Installer.
![image](https://user-images.githubusercontent.com/1292492/212529484-f6d7fd7e-3205-4cba-9433-4a9e16ad3538.png)

I'll amend this post with an answer to your Release Zip.
----
```
--------------------------------
08:14:54.342 | ERR [LOCALPLUGIN] Error while loading Resizable HUD
System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation.
 ---> System.NullReferenceException: Object reference not set to an instance of an object.
   at ResizableHUD.Plugin..ctor(DalamudPluginInterface pi, CommandManager commands, ChatGui chat, ClientState clientState)
   at System.RuntimeMethodHandle.InvokeMethod(Object target, Void** arguments, Signature sig, Boolean isConstructor)
   at System.Reflection.ConstructorInvoker.Invoke(Object obj, IntPtr* args, BindingFlags invokeAttr)
   --- End of inner exception stack trace ---
   at System.Reflection.ConstructorInvoker.Invoke(Object obj, IntPtr* args, BindingFlags invokeAttr)
   at System.Reflection.RuntimeConstructorInfo.Invoke(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
   at Dalamud.IoC.Internal.ServiceContainer.CreateAsync(Type objectType, Object[] scopedObjects) in C:\goatsoft\companysecrets\dalamud\IoC\Internal\ServiceContainer.cs:line 107
   at Dalamud.Plugin.Internal.Types.LocalPlugin.LoadAsync(PluginLoadReason reason, Boolean reloading) in C:\goatsoft\companysecrets\dalamud\Plugin\Internal\Types\LocalPlugin.cs:line 456
--------------------------------
08:14:54.342 | ERR [PLUGINW] Plugin installer threw an unexpected error
System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation.
 ---> System.NullReferenceException: Object reference not set to an instance of an object.
   at ResizableHUD.Plugin..ctor(DalamudPluginInterface pi, CommandManager commands, ChatGui chat, ClientState clientState)
   at System.RuntimeMethodHandle.InvokeMethod(Object target, Void** arguments, Signature sig, Boolean isConstructor)
   at System.Reflection.ConstructorInvoker.Invoke(Object obj, IntPtr* args, BindingFlags invokeAttr)
   --- End of inner exception stack trace ---
   at System.Reflection.ConstructorInvoker.Invoke(Object obj, IntPtr* args, BindingFlags invokeAttr)
   at System.Reflection.RuntimeConstructorInfo.Invoke(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
   at Dalamud.IoC.Internal.ServiceContainer.CreateAsync(Type objectType, Object[] scopedObjects) in C:\goatsoft\companysecrets\dalamud\IoC\Internal\ServiceContainer.cs:line 107
   at Dalamud.Plugin.Internal.Types.LocalPlugin.LoadAsync(PluginLoadReason reason, Boolean reloading) in C:\goatsoft\companysecrets\dalamud\Plugin\Internal\Types\LocalPlugin.cs:line 456
```
Same failure.

**Comment by Drahsid** [comment]
Thanks, I'll investigate this and update you when I fix it.

**Comment by Nox13last** [comment]
I feel this is a significant thing to add. For a pure testing environment, I deleted and repopulated dalamudConfig.json and deleted \installedPlugins\ and \pluginConfig\. There should be absolutely no problems at this point. No conflicting plugins.
I have also attempted to load the plugin both from the splash screen (before the Character Select) (if I have to hear "tales of loss and fire and faaaiith" _one more **goddamn** time_...), and from in-game (positioned at my Island Sanctuary).

The plugin failed. Both in attempting to get it from the repo, and from the ZIP you've provided me, and in every circumstance I've been able to test it in.
I don't know about the OP (I _did_ commandeer this ticket), but I am able to provide further information on the environment the game is running in, on request.

**Comment by Drahsid** [comment]
I believe the issue is with the automation that I wrote to upload the plugin.  To confirm, check if this works: [net7.0-windows.zip](https://github.com/Drahsid/ResizableHUD/files/10419893/net7.0-windows.zip)

**Comment by Nox13last** [comment]
It did not work.
But for what it's worth, the log snippet reports that it did not work _but in a new and slightly different way._ 🎉
```
--------------------------------
10:13:02.061 | INF [LOCALPLUGIN] Loading ResizableHUD.dll
--------------------------------
10:13:02.080 | ERR [LOCALPLUGIN] Error while loading Resizable HUD
System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation.
 ---> System.NullReferenceException: Object reference not set to an instance of an object.
   at ResizableHUD.Plugin..ctor(DalamudPluginInterface pi, CommandManager commands, ChatGui chat, ClientState clientState) in F:\projs\ResizableHUD\Plugin.cs:line 63
   at System.RuntimeMethodHandle.InvokeMethod(Object target, Void** arguments, Signature sig, Boolean isConstructor)
   at System.Reflection.ConstructorInvoker.Invoke(Object obj, IntPtr* args, BindingFlags invokeAttr)
   --- End of inner exception stack trace ---
   at System.Reflection.ConstructorInvoker.Invoke(Object obj, IntPtr* args, BindingFlags invokeAttr)
   at System.Reflection.RuntimeConstructorInfo.Invoke(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
   at System.Reflection.MethodBase.Invoke(Object obj, Object[] parameters)
   at Dalamud.IoC.Internal.ServiceContainer.CreateAsync(Type objectType, Object[] scopedObjects) in C:\goatsoft\companysecrets\dalamud\IoC\Internal\ServiceContainer.cs:line 104
   at Dalamud.Plugin.Internal.Types.LocalPlugin.LoadAsync(PluginLoadReason reason, Boolean reloading) in C:\goatsoft\companysecrets\dalamud\Plugin\Internal\Types\LocalPlugin.cs:line 422
--------------------------------
10:13:02.080 | ERR [PLUGINW] Plugin installer threw an unexpected error
System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation.
 ---> System.NullReferenceException: Object reference not set to an instance of an object.
   at ResizableHUD.Plugin..ctor(DalamudPluginInterface pi, CommandManager commands, ChatGui chat, ClientState clientState) in F:\projs\ResizableHUD\Plugin.cs:line 63
   at System.RuntimeMethodHandle.InvokeMethod(Object target, Void** arguments, Signature sig, Boolean isConstructor)
   at System.Reflection.ConstructorInvoker.Invoke(Object obj, IntPtr* args, BindingFlags invokeAttr)
   --- End of inner exception stack trace ---
   at System.Reflection.ConstructorInvoker.Invoke(Object obj, IntPtr* args, BindingFlags invokeAttr)
   at System.Reflection.RuntimeConstructorInfo.Invoke(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
   at System.Reflection.MethodBase.Invoke(Object obj, Object[] parameters)
   at Dalamud.IoC.Internal.ServiceContainer.CreateAsync(Type objectType, Object[] scopedObjects) in C:\goatsoft\companysecrets\dalamud\IoC\Internal\ServiceContainer.cs:line 104
   at Dalamud.Plugin.Internal.Types.LocalPlugin.LoadAsync(PluginLoadReason reason, Boolean reloading) in C:\goatsoft\companysecrets\dalamud\Plugin\Internal\Types\LocalPlugin.cs:line 422
```

**Comment by Drahsid** [comment]
Try it from the repo now, I just pushed an update which should have fixed it.

**Comment by Nox13last** [comment]
Failed.
Furthermore, I wasn't sure until closely examining the log snippet as to whether or not I received fresh files. The version number on the install button appeared to be identical to before. If you believe there's no reason to change the version number between attempts, that's fine, and any assurances from you that the files are different will have to suffice. But I had to double-check on the GH repository to see if files had indeed been changed.
```
--------------------------------
10:59:32.326 | INF [PLUGINM] Installed plugin Resizable HUD (testing=False)
--------------------------------
10:59:32.326 | INF [PLUGINM] Loading plugin Resizable HUD
--------------------------------
10:59:32.329 | INF [LOCALPLUGIN] Loading ResizableHUD.dll
--------------------------------
10:59:32.331 | ERR [LOCALPLUGIN] Error while loading Resizable HUD
System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation.
 ---> System.NullReferenceException: Object reference not set to an instance of an object.
   at ResizableHUD.Plugin..ctor(DalamudPluginInterface pi, CommandManager commands, ChatGui chat, ClientState clientState)
   at System.RuntimeMethodHandle.InvokeMethod(Object target, Void** arguments, Signature sig, Boolean isConstructor)
   at System.Reflection.ConstructorInvoker.Invoke(Object obj, IntPtr* args, BindingFlags invokeAttr)
   --- End of inner exception stack trace ---
   at System.Reflection.ConstructorInvoker.Invoke(Object obj, IntPtr* args, BindingFlags invokeAttr)
   at System.Reflection.RuntimeConstructorInfo.Invoke(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
   at System.Reflection.MethodBase.Invoke(Object obj, Object[] parameters)
   at Dalamud.IoC.Internal.ServiceContainer.CreateAsync(Type objectType, Object[] scopedObjects) in C:\goatsoft\companysecrets\dalamud\IoC\Internal\ServiceContainer.cs:line 104
   at Dalamud.Plugin.Internal.Types.LocalPlugin.LoadAsync(PluginLoadReason reason, Boolean reloading) in C:\goatsoft\companysecrets\dalamud\Plugin\Internal\Types\LocalPlugin.cs:line 422
--------------------------------
10:59:32.331 | ERR [PLUGINW] Plugin installer threw an unexpected error
System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation.
 ---> System.NullReferenceException: Object reference not set to an instance of an object.
   at ResizableHUD.Plugin..ctor(DalamudPluginInterface pi, CommandManager commands, ChatGui chat, ClientState clientState)
   at System.RuntimeMethodHandle.InvokeMethod(Object target, Void** arguments, Signature sig, Boolean isConstructor)
   at System.Reflection.ConstructorInvoker.Invoke(Object obj, IntPtr* args, BindingFlags invokeAttr)
   --- End of inner exception stack trace ---
   at System.Reflection.ConstructorInvoker.Invoke(Object obj, IntPtr* args, BindingFlags invokeAttr)
   at System.Reflection.RuntimeConstructorInfo.Invoke(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
   at System.Reflection.MethodBase.Invoke(Object obj, Object[] parameters)
   at Dalamud.IoC.Internal.ServiceContainer.CreateAsync(Type objectType, Object[] scopedObjects) in C:\goatsoft\companysecrets\dalamud\IoC\Internal\ServiceContainer.cs:line 104
   at Dalamud.Plugin.Internal.Types.LocalPlugin.LoadAsync(PluginLoadReason reason, Boolean reloading) in C:\goatsoft\companysecrets\dalamud\Plugin\Internal\Types\LocalPlugin.cs:line 422
   at Dalamud.Plugin.Internal.PluginManager.LoadPluginAsync(FileInfo dllFile, LocalPluginManifest manifest, PluginLoadReason reason, Boolean isDev, Boolean isBoot, Boolean doNotLoad) in C:\goatsoft\companysecrets\dalamud\Plugin\Internal\PluginManager.cs:line 871
   at Dalamud.Plugin.Internal.PluginManager.InstallPluginAsync(RemotePluginManifest repoManifest, Boolean useTesting, PluginLoadReason reason) in C:\goatsoft\companysecrets\dalamud\Plugin\Internal\PluginManager.cs:line 818
```
Edit, t+40 minutes.
I glanced over at the Plugin Installer and saw ResizableHUD version 1.0.0.**11546**
Exact same error log as above.

**Comment by Varstahl** [comment]
> When asking for help with plugins not loading properly, it's generally a good idea to include a log. You can get it by typing `/xllog` in chat, and then pressing the "copy to clipboard" button.

Sorry, I should've started with that, but I'm very new to all of this.

> Try it from the repo now, I just pushed an update which should have fixed it.

I uninstalled and installed Resizable HUD v1.0.0.14273 straight from the repo, it works perfectly now, sorry for the lack of logs :)

**Comment by Drahsid** [comment]
Just want to update @Nox13last here, the problem seems to have been solved. It looks like the zip files on my server were wrong, so that's on me. In other news, I did find a different bug which I fixed.

**Comment by Varstahl** [comment]
@Drahsid just to know, the bug you fixed is it related to #2, with an update pending? Or is it something else entirely?

**Comment by Nox13last** [comment]
> Just want to update @Nox13last here, the problem seems to have been solved. It looks like the zip files on my server were wrong, so that's on me. In other news, I did find a different bug which I fixed.

Ay, all's well that ends well. Glad we got there. 
![image](https://user-images.githubusercontent.com/1292492/212564697-503d8c82-16c3-458a-87f0-edd8c00fff5f.png)

> > When asking for help with plugins not loading properly, it's generally a good idea to include a log. You can get it by typing `/xllog` in chat, and then pressing the "copy to clipboard" button.
> 
> Sorry, I should've started with that, but I'm very new to all of this.

It's all fine. Sorry if my advice came across as curt. Plugin developers really love logs - they're infinitely better than _"plogon no work, make fix!",_ especially if they're having difficulty reproducing the problem themselves.

I'll give you fine people your ticket back now. 👋
